### ISSUE TITLE
Potential conflict between invalid autocomplete attributes and screen readers

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5352]
As part of #5136, the date / datetime fields now use invalid `autocomplete` attributes to disable autocompletion in Chrome. This causes Axe to raise the following issue:

> autocomplete attribute must be used correctly (https://dequeuniversity.com/rules/axe/3.2/autocomplete-valid?application=axeAPI)

> Failure to provide autocomplete values in form fields results in inaccessible content. Screen readers do not read identified autocomplete form fields if the appropriate autocomplete attribute values are missing. Users cannot correctly navigate forms when screen readers cannot provide adequate information to the user regarding form field interaction requirements.

> The purpose for each common input field that collects an individual's personal data is programmatically defined based on the list of 53 Input Purposes for User Interface Components. The autocomplete attribute values must be valid and correctly applied for screen readers to function correctly.

According to the spec (https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofilling-form-controls:-the-autocomplete-attribute), `autocomplete` must either be `off`, `on`, or any of the predefined autofill values.

I'm not entirely sure what the consequences of using invalid values are for screen readers, this needs further investigation.
